external/mmdetection/detection_tasks/apis/detection/inference_task.py

```diff
@@ -306,25 +319,35 @@ def _infer_detector(model: torch.nn.Module, config: Config, dataset: DatasetEnti
 
         eval_predictions = []
         feature_vectors = []
+        saliency_maps = []
 
         def dump_features_hook(mod, inp, out):
             with torch.no_grad():
-                feature_map = out[-1]
-                feature_vector = torch.nn.functional.adaptive_avg_pool2d(feature_map, (1, 1))
+                feature_vector = get_feature_vector(out)
                 assert feature_vector.size(0) == 1
             feature_vectors.append(feature_vector.view(-1).detach().cpu().numpy())
 
         def dummy_dump_features_hook(mod, inp, out):
             feature_vectors.append(None)
 
-        hook = dump_features_hook if dump_features else dummy_dump_features_hook
+        def dump_saliency_hook(mod, inp, out):
+            with torch.no_grad():
+                saliency_map = get_saliency_map(out[0])
```

Comment:
Same question: out[0] or out[-1]?
Or am I misunderstanding the index? Is it layer index?

Reply:
I'm currently taking the largest feature map at index 0 i.e. `out[0]`.

Reply:
As I commented in the mmdet PR, let's use out[-1] which has most high-level semantic.

Reply:
Updated